compostables/useAuthValidator.ts

```diff
@@ -0,0 +1,39 @@
+import { any, z } from 'zod'
+
+export default function useAuthValidator(email: string, password: string, type: string) {
+  let body
+
+  try {
+    if (type == "login") {
+      const loginSchema = z.object({
+        email: z.string().email(),
+        password: z.string().min(6),
+      })
+      body = loginSchema.parse({ email, password })
+    } else if (type == "signup") {
+      const signUpSchema = z.object({
+        email: z.string().email(),
+        password: z.string().min(6),
+      })
+      body = signUpSchema.parse({ email, password })
+    }
+    return {
+      flag: true,
+      ...body,
+    }
+  } catch (err) {
+    let errors = {
+      flag: false,
+    }
+
+    if (err instanceof z.ZodError) {
+      const zodIssues = err.issues
+
+      for (let i = 0; i < zodIssues.length; i++) {
+        // errors[zodIssues[i].path[0]] = zodIssues[i].message
```

Comment:
Is this needed? My guess it that we want to uncomment it so that we can actually return error messages on failed logins.